main/autogen/autogen.cc

```diff
@@ -663,15 +664,116 @@ string ParsedFile::toMsgpack(core::Context ctx, int version) {
     return write.pack(ctx, *this);
 }
 
-void ParsedFile::classlist(core::Context ctx, vector<string> &out) {
-    auto nameToString = [&](const auto &nm) -> string { return nm.data(ctx)->show(ctx); };
+vector<string> ParsedFile::listAllClasses(const core::Context ctx) {
+    vector<string> out;
+
     for (auto &def : defs) {
         if (def.type != Definition::Class) {
             continue;
         }
         auto names = showFullName(ctx, def.id);
-        out.emplace_back(fmt::format("{}", fmt::map_join(names, "::", nameToString)));
+        out.emplace_back(fmt::format("{}", fmt::map_join(names, "::", [&ctx](const core::NameRef &nm) -> string {
+                                         return nm.data(ctx)->show(ctx);
+                                     })));
+    }
+
+    return out;
+}
+
+optional<AutogenSubclassMap> ParsedFile::listAllSubclasses(const core::Context ctx,
+                                                           const vector<string> &absolutePathsToIgnore,
+                                                           const vector<string> &relativePathsToIgnore) {
+    // We prepend "/" to `path` to mimic how `isFileIgnored` gets called elsewhere
+    if (sorbet::FileOps::isFileIgnored("", fmt::format("/{}", path), absolutePathsToIgnore, relativePathsToIgnore)) {
+        return nullopt;
+    }
+
+    AutogenSubclassMap out;
+
+    for (const Reference &ref : refs) {
+        DefinitionRef defn = ref.parent_of;
+
+        if (!defn.exists()) {
```

Comment:
why are there `defn`s that dont `exist` in this list?

Reply:
This list includes `Reference`s that have no `parent_of` (e.g., `module Foo; end`), which I don't care about.